packages/expo-notifications/src/InstallationIdProvider.web.ts

```diff
@@ -0,0 +1,21 @@
+import UUID from 'uuid-js';
+
+import { InstallationIdProvider } from './InstallationIdProvider';
+
+const INSTALLATION_ID_KEY = '@@expo-notifications.InstallationId@@';
+
+export default {
+  getInstallationIdAsync: async () => {
+    let installationId = localStorage.getItem(INSTALLATION_ID_KEY);
```

Comment:
localStorage can fail for several reasons. expo-constants does this:

```
    let installationId;
    try {
      installationId = localStorage.getItem(ID_KEY);
      if (installationId == null || typeof installationId !== 'string') {
        installationId = UUID.create().toString();
        localStorage.setItem(ID_KEY, installationId as string);
      }
    } catch (error) {
      installationId = _sessionId;  // <----- this is just a UUID that is generated for the duration of the page's lifetime
    }
    return installationId;
```

Reply:
Went with this ⬆️ + local _session ID_.

https://github.com/expo/expo/blob/632138f5cbf3ddbbbdb609b2b1e260f61b5161e4/packages/expo-notifications/src/InstallationIdProvider.web.ts#L7-L28

I'm torn between using a lazy getter for this fallback installation ID and just calculating and assigning it on every run.